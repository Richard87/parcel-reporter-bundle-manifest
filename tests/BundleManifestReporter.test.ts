import { exec as execBeforePromisify } from 'child_process'
import util from 'util'
import fs from 'fs'
import path from 'path'

const exec = util.promisify(execBeforePromisify)

jest.setTimeout(120000)

beforeAll(() => {
  return exec('npm run build')
})

test('parcel-alpha', async () => {
  process.chdir(path.join(__dirname, './fixtures/parcel-alpha'))
  await buildAndAssertManifestFile()
})

test('parcel-nightly', async () => {
  process.chdir(path.join(__dirname, './fixtures/parcel-nightly'))
  await buildAndAssertManifestFile()
})

async function buildAndAssertManifestFile() {
  const { stderr } = await exec('rm -rf .parcel-cache dist node_modules && npm install && npm run build')
  if (stderr) {
    throw new Error(stderr)
  }
  const parcelManifest = JSON.parse(fs.readFileSync('./dist/parcel-manifest.json').toString())
  expect(parcelManifest).toMatchSnapshot()
}
