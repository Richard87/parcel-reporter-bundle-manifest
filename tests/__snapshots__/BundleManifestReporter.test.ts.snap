// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`parcel-alpha 1`] = `
Object {
  "index.html": "index.html",
  "script.js": "script.1212fd4f.js",
}
`;

exports[`parcel-nightly 1`] = `
Object {
  "index.html": "index.html",
  "script.js": "script.b1e49460.js",
}
`;
